client/src/components/DateField/DateField.js

```diff
@@ -63,21 +73,20 @@ class DateField extends TextField {
     const enteredValue = event.target.value;
     let isoValue = '';
 
-    // When browser support input=date the date value is already in iso format
-    // and html5 is enabled
-    if (!this.props.data.html5 || (this.hasNativeSupport() && this.props.data.html5)) {
+    // When browser support input=date the date value is already in iso format and html5 is enabled
+    if (this.asHTML5()) {
       isoValue = enteredValue;
     } else {
       isoValue = this.convertToIso(enteredValue);
     }
 
     if (typeof this.props.onChange === 'function') {
-      this.triggerChange(isoValue);
+      this.triggerChange(event, isoValue);
     }
   }
 
-  triggerChange(value) {
-    this.props.onChange(value);
+  triggerChange(event, value) {
+    this.props.onChange(event, { id: this.props.id, value });
```

Comment:
Apparently all of these onChange() events fail if you don't pass in `value`. Will need to review each of these tomorrow.

Reverting them fixes it.

Reply:
See redux forms docs on onChange here https://redux-form.com/7.1.2/docs/api/field.md/#-code-input-onchange-eventorvalue-function-code-

Reply:
hmm, something else is wrong then, because it should be `this.props.onChange(event, { id: this.props.id, value });`
other form fields follow this pattern and return two objects.